From e3d66ddeba01bce34036bb521144d291793a31a9 Mon Sep 17 00:00:00 2001
From: wj <545818496@qq.com>
Date: Fri, 10 May 2024 17:31:22 +0800
Subject: [PATCH] sdl bpp=16

---
 RetroFE/Source/SDL.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/RetroFE/Source/SDL.cpp b/RetroFE/Source/SDL.cpp
index 29febe9..43ec289 100755
--- a/RetroFE/Source/SDL.cpp
+++ b/RetroFE/Source/SDL.cpp
@@ -164,7 +164,7 @@ bool SDL::initialize( Configuration &config )
 
         Logger::write( Logger::ZONE_INFO, "SDL", ss.str( ));
 
-        window_ = SDL_SetVideoMode(windowWidth_, windowHeight_, 32, windowFlags);
+        window_ = SDL_SetVideoMode(windowWidth_, windowHeight_, 16, windowFlags);
         if ( window_ == NULL )
         {
             std::string error = SDL_GetError( );
@@ -187,7 +187,7 @@ bool SDL::initialize( Configuration &config )
         bmask = 0x00ff0000;
         amask = 0xff000000;
 #endif*/
-        window_virtual_ = SDL_CreateRGBSurface(0, windowWidth_, windowHeight_, 32, 0,0,0,0);
+        window_virtual_ = SDL_CreateRGBSurface(0, windowWidth_, windowHeight_, 16, 0,0,0,0);
         //window_virtual_ = SDL_CreateRGBSurface(0, windowWidth_, windowHeight_, 32, rmask, gmask, bmask, amask); // colors are reversed with this !
         if ( window_virtual_ == NULL )
         {
@@ -456,8 +456,8 @@ void SDL::ditherSurface32bppTo16Bpp(SDL_Surface * src_surface){
 // Copy virtual window to HW window and Flip display
 void SDL::renderAndFlipWindow( )
 {
-	//SDL_BlitSurface(window_virtual_, NULL, window_, NULL);
-    memcpy(window_->pixels, window_virtual_->pixels, window_->h*window_->w*sizeof(uint32_t));
+	SDL_BlitSurface(window_virtual_, NULL, window_, NULL);
+    //memcpy(window_->pixels, window_virtual_->pixels, window_->h*window_->w*sizeof(uint32_t));
 	//SDL_Rotate_270(window_virtual_, window_);
 
 	SDL_Flip(window_);
-- 
2.25.1

